buildSrc/src/main/kotlin/TargetsConfig.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+@file:Suppress("UNUSED_VARIABLE")
+
+import org.gradle.api.*
+import org.gradle.kotlin.dsl.*
+import org.jetbrains.kotlin.gradle.targets.native.tasks.*
+
+fun Project.configureTargets() {
+    val coroutines_version: String by extra
+    val kotlin_version: String by extra
+
+    val files = project.projectDir.listFiles() ?: emptyArray()
+
+    val hasCommon = files.any { it.name == "common" }
+    val hasJvmAndNix = hasCommon || files.any { it.name == "jvmAndNix" }
+    val hasPosix = hasCommon || files.any { it.name == "posix" }
+    val hasDesktop = hasPosix || files.any { it.name == "desktop" }
+    val hasNix = hasPosix || files.any { it.name == "nix" }
+    val hasDarwin = hasNix || files.any { it.name == "darwin" }
+    val hasJs = hasCommon || files.any { it.name == "js" }
+    val hasJvm = hasCommon || files.any { it.name == "jvm" }
+
+    kotlin {
+        if (hasJvm) {
+            jvm()
+            configureJvm()
+        }
+
+        if (hasJs) {
+            js {
+                nodejs()
+                browser()
+            }
+
+            configureJs()
+        }
+
+        if (hasPosix) {
```

Comment:
is it expected that those targets will be created even if `IDEA_ACTIVE` is false?

Reply:
minor: I see that `posixTargets` also called down the file (so similar as with JVM double install of targets, even if should work, better to remove it from here for consistency, as this file is already hard to understand in detail

Reply:
`posixTargets()` is just an accessor POSIX targets(which create them if they don't exist).

I would say `posixTargets()` is called here for consistency: we create all targets before getting down to the source set configuration